app/lib/audit/backend.dart

```diff
@@ -93,14 +100,34 @@ class AuditBackend {
     );
   }
 
+  @visibleForTesting
+  String nextTimestamp(DateTime last, DateTime next) {
+    final nextDayStart =
+        DateTime.utc(next.year, next.month, next.day).add(Duration(days: 1));
+    return nextDayStart.isBefore(last) && nextDayStart.isAfter(next)
+        ? nextDayStart.toIso8601String().split('T').first
+        : next.toIso8601String();
+  }
+
   /// Parses the `before` query parameter and returns the parsed timestamp.
   ///
   /// Returns a timestamp slightly into the future if the parameter is missing.
   /// Returns null if the query parameter is invalid.
```

Comment:
```suggestion
  /// Returns `null` if the query parameter is invalid.
```

Why not just throw an `ResponseException` with `400`?